include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -119,6 +180,44 @@ template<typename Signed>
 [[vk::ext_instruction( spv::OpBitFieldSExtract )]]
 Signed bitFieldSExtract( Signed val, uint32_t offsetBits, uint32_t numBits );
 
+}
+
+// should be in `nbl::hlsl::spirv` but would collide right now
+namespace experimental
+{
+using namespace nbl::hlsl::spirv;
+
+template<typename T, typename U>
+[[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
+[[vk::ext_instruction(spv::OpBitcast)]]
+T bitcast(U);
+
+template<typename T, uint32_t StorageClass>
+[[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
+[[vk::ext_instruction(spv::OpBitcast)]]
+uint64_t bitcast(pointer_t<StorageClass,T>);
+
+template<typename T, uint32_t StorageClass>
+[[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
+[[vk::ext_instruction(spv::OpBitcast)]]
+pointer_t<StorageClass,T> bitcast(uint64_t);
+
+template<typename T, typename U, uint32_t StorageClass>
+[[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
+[[vk::ext_instruction(spv::OpBitcast)]]
+pointer_t<StorageClass,T> bitcast(pointer_t<StorageClass,U>);
+}
+
+namespace atomic64 {
+template<typename T> // integers operate on 2s complement so same op for signed and unsigned
+[[vk::ext_capability(spv::CapabilityInt64Atomics)]]
+[[vk::ext_instruction(spv::OpAtomicIAdd)]]
+T atomicIAdd([[vk::ext_reference]] T ptr, uint32_t memoryScope, uint32_t memorySemantics, T value);
+
+template<typename T, typename Ptr_T> // DXC Workaround
+[[vk::ext_capability(spv::CapabilityInt64Atomics)]]
+[[vk::ext_instruction(spv::OpAtomicIAdd)]]
+typename nbl::hlsl::enable_if<is_spirv_type_v<Ptr_T>, T>::type atomicIAdd(Ptr_T ptr, uint32_t memoryScope, uint32_t memorySemantics, T value);
 }
```

Comment:
use `enable_if` conditioned on `is_same_v<T,uint64_t>||is_same_v<T,int64_t>` to stop the overloads colliding

Reply:
Done